packages/dashmate/src/core/startCoreFactory.js

```diff
@@ -34,19 +34,25 @@ function startCoreFactory(
     const coreCommand = [
       'dashd',
     ];
+
     const isMasternode = config.get('core.masternode.enable');
-    const operatorKey = config.get('core.masternode.operator.privateKey');
 
-    if (options.wallet) {
-      coreCommand.push('--disablewallet=0');
-    } else if (isMasternode && Boolean(operatorKey)) {
-      coreCommand.push(`-masternodeblsprivkey=${operatorKey}`);
+    if (isMasternode) {
+      // Check operatorPrivateKey is set
+      config.get('core.masternode.operator.privateKey', true);
     }
 
     if (options.addressIndex) {
       coreCommand.push('--addressindex=1');
     }
 
+    if (options.wallet) {
+      config.set('core.masternode.enable', 0)
```

Comment:
Did you test it? It's not gonna work because dashd.conf already rendered with key. I guess you need to render additional config and pass it specifically here or use dockerode to create container, pass required command option and mount a volume. Both are not  trivial